changelog/add_traits_hasAliasing.dd

```diff
@@ -0,0 +1,20 @@
+Add new `__traits(hasAliasing, T)`.
+
+A call `__traits(hasAliasing, T)` is `true` if and only if the type `T` has any
+non-`immutable` indirections.
```

Comment:
Yes, very good point. I lean more torwards `hasMutableIndirections` to make the difference with hasIndirections completely obvious.

Reply:
One problem I see with using the naming `hasMutableAliasing` is misinterpreting it as meaning non-`const` aliasing.

Reply:
So, what about renaming this builtin trait from hasAliasing to hasMutableIndirections? Ping, @andralex @RazvanN7?

Reply:
If we take a look at phobos, we have `hasIndirections` and `hasAliasing`. At a first glance the difference is not obvious between them. If we are to follow the phobos example, `hasAliasing` is fine in this case. If we change the name here, then we will be forced to update phobos to be consistent. Either way seems icky to me, since both `hasAliasing` and `hasIndirections` don't have any indication in their name with regards to mutability.

If we assume that `hasAliasing` and `hasIndirections` have been properly absorbed by the users, we can go with `traits(hasAliasing)`.